src/textUtils.c

```diff
@@ -139,6 +182,19 @@ void str_validateTextBuffer(const uint8_t* text, size_t textSize)
 	}
 }
 
+// check if it is printable ASCII between 33 and 126
+bool str_isTextPrintable(const uint8_t* text, size_t textSize)
```

Comment:
I find it a bit unsettling that this function would not work properly for null-terminated strings, not sure though what I would do differently - maybe rename it to `str_isAsciiPrintableBuffer()` to convey that the function does not really expect a proper C string